markdown/Cheapskate/Inlines.hs

```diff
@@ -415,15 +415,14 @@ pAutolink = do
          | otherwise   -> fail "Unknown contents of <>"
 
 autoLink :: Text -> Inlines
-autoLink t = singleton $ Link (toInlines t) (Url t) (T.empty)
-  where toInlines t' = case parse pToInlines t' of
-                         Right r   -> r
-                         Left e    -> error $ "autolink: " ++ show e
-        pToInlines = mconcat <$> many strOrEntity
+autoLink t = 
+  case parse pToInlines t of
+      Right _   -> singleton $ Autolink t
+      Left e    -> error $ "autolink: " ++ show e
```

Comment:
To be honest, I fail to think of a case where this would fail thinking
So I don't full understand this part of the code.